src/plugins/discourse/weights.js

```diff
@@ -1,9 +1,75 @@
 // @flow
 
 import {type NodeWeight} from "../../core/weights";
+import * as C from "../../util/combo";
+import * as NullUtil from "../../util/null";
 import {DEFAULT_TRUST_LEVEL_TO_WEIGHT} from "./createGraph";
 import {type User} from "./fetch";
 
+export type SerializedConfig = {|
+  +defaultWeight?: NodeWeight,
+  +weights?: {
+    [string]: NodeWeight,
+  },
+|};
+
+export type Config = {|
+  +defaultWeight: NodeWeight,
+  +weights: {
+    [string]: NodeWeight,
+  },
+|};
+
+export type CategoryConfig = Config;
+export type TagConfig = Config;
+
+export const DEFAULT_CONFIG: Config = {
+  defaultWeight: 1,
+  weights: {},
+};
+
+export function upgrade(c: SerializedConfig): Config {
+  return {
+    defaultWeight: NullUtil.orElse(
+      c.defaultWeight,
+      DEFAULT_CONFIG.defaultWeight
+    ),
+    weights: NullUtil.orElse(c.weights, DEFAULT_CONFIG.weights),
+  };
```

Comment:
Perhaps just

```javascript
function upgrade(c: SerializedConfig): Config {
  return {
    defaultWeight: 1,
    weights: {},
    ...config,
  }
}
```

? More readable since it’s shorter, only cares about presence of fields
rather than nullity/undefinedness, and makes it easy to inline the
`DEFAULT_CONFIG` object entirely.